### PR TITLE
Add meaningful resource strings for System.Web.Services

### LINKED ISSUE #11623: System.Web.Services does not have useful errors
System.Web.Services uses the resource string IDs instead of the values.

https://github.com/mono/mono/blob/c5b88ec4f323f2bdb7c7d0a595ece28dae66579c/mcs/class/System.Web.Services/ReferenceSources/Res.cs

This means exceptions have messages like `"ThereWasAnErrorDownloading0"` instead of `"There was an error downloading 'https:/example.com/someurl.asmx'"`

### PR DESCRIPTION
**Pull request by mhutch**
Generated from mcs/class/referencesource/System.Web.Services/System.Web.Services.txt using resx2sr

Also adds support for txt files and custom class names to resx2sr

Fixes #11623

### CODE REVIEW

**Comment by luhenry** [comment]
@monojenkins squash

**Comment by luhenry** [comment]
@monojenkins build failed

File: mcs/tools/resx2sr/resx2sr.cs [hunk 1]
```diff
@@ -111,13 +119,24 @@ static void GenerateFile (List<Tuple<string, string, string>> txtStrings, CmdOpt
 			str.WriteLine ("//");
 			str.WriteLine ();
 
-			str.WriteLine ("partial class SR");
+			int nsIdx = className.LastIndexOf('.');
```
Comment (marek-safar) [inline mcs/tools/resx2sr/resx2sr.cs]: Would be great to keep the parens formatting same for the changes

**Comment by luhenry** [comment]
@monojenkins commit csproj

File: mcs/tools/resx2sr/resx2sr.cs (hunk 1 above)
Comment (mhutch) [reply mcs/tools/resx2sr/resx2sr.cs]:
```suggestion
			int nsIdx = className.LastIndexOf ('.');
```

Comment (mhutch) [inline mcs/tools/resx2sr/resx2sr.cs]:
```suggestion
			int nsIdx = className.LastIndexOf ('.');
```

File: mcs/tools/resx2sr/resx2sr.cs [hunk 2]
```diff
@@ -111,13 +119,24 @@ static void GenerateFile (List<Tuple<string, string, string>> txtStrings, CmdOpt
...
+			if (nsIdx > 0) {
+				str.WriteLine ($"namespace {className.Substring(0,nsIdx)}");
+				str.WriteLine ("{");
+				className = className.Substring(nsIdx+1);
```
Comment (mhutch) [inline mcs/tools/resx2sr/resx2sr.cs]:
```suggestion
				className = className.Substring (nsIdx+1);
```

File: mcs/tools/resx2sr/resx2sr.cs [hunk 3]
```diff
@@ -111,13 +119,24 @@ static void GenerateFile (List<Tuple<string, string, string>> txtStrings, CmdOpt
 			str.WriteLine ("//");
 			str.WriteLine ();
 
-			str.WriteLine ("partial class SR");
+			int nsIdx = className.LastIndexOf('.');
+			if (nsIdx > 0) {
+				str.WriteLine ($"namespace {className.Substring(0,nsIdx)}");
```
Comment (mhutch) [inline mcs/tools/resx2sr/resx2sr.cs]:
```suggestion
				str.WriteLine ($"namespace {className.Substring (0, nsIdx)}");
```

File: mcs/tools/resx2sr/resx2sr.cs [hunk 4]
```diff
@@ -111,13 +119,24 @@ static void GenerateFile (List<Tuple<string, string, string>> txtStrings, CmdOpt
...
+			}
+
+			str.WriteLine ($"partial class {className}");
 			str.WriteLine ("{");
 
 			var dict = new Dictionary<string, string> ();
 
 			foreach (var entry in txtStrings) {
 
+				if (options.ExistingOnly && !ExistingKeys.ContainsKey(entry.Item1)) {
```
Comment (mhutch) [inline mcs/tools/resx2sr/resx2sr.cs]:
```suggestion
				if (options.ExistingOnly && !ExistingKeys.ContainsKey (entry.Item1)) {
```

File: mcs/tools/resx2sr/resx2sr.cs [hunk 5]
```diff
@@ -156,23 +186,42 @@ static bool LoadStrings (List<Tuple<string, string, string>> resourcesStrings, L
...
+		foreach (var line in File.ReadAllLines (fileName)) {
+			if (line.Length == 0 || line[0] == ';') {
+				continue;
+			}
+			var idx = line.IndexOf ('=');
+			if (idx < 1) {
+				Console.Error.WriteLine ($"Error reading resource file '{fileName}'");
+				continue;
+			}
+			yield return Tuple.Create (line.Substring(0, idx), line.Substring(idx+1), (string)null);
```
Comment (mhutch) [inline mcs/tools/resx2sr/resx2sr.cs]:
```suggestion
			yield return Tuple.Create (line.Substring (0, idx), line.Substring (idx+1), (string)null);
```

**Comment by luhenry** [comment]
@monojenkins build

**Comment by baulig** [comment]
Could you please at least _test_ your changes prior to committing next time?

The minimum thing you need to do when touching that tool is to do a `make update-corefx-sr` in corlib, then type `make` to ensure that it still builds and do a `git diff` to look at what you have done.

File: mcs/tools/resx2sr/resx2sr.cs [hunk 6]
```diff
@@ -136,7 +155,18 @@ static void GenerateFile (List<Tuple<string, string, string>> txtStrings, CmdOpt
 
 				str.WriteLine ();
 			}
+
```
Comment (baulig) [inline mcs/tools/resx2sr/resx2sr.cs]: This change is wrong and broke the tool - and it's not just the cosmetic issue of not adding newlines here, which makes the output look _extremely ugly_.

**Comment by baulig** [comment]
Commented inline on the section of the code which broke things.

**Pull request merged**